datafusion/optimizer/src/eliminate_filter.rs

```diff
@@ -78,17 +56,47 @@ impl OptimizerRule for EliminateFilter {
     fn apply_order(&self) -> Option<ApplyOrder> {
         Some(ApplyOrder::TopDown)
     }
+
+    fn supports_rewrite(&self) -> bool {
+        true
+    }
+
+    fn rewrite(
+        &self,
+        plan: LogicalPlan,
+        _config: &dyn OptimizerConfig,
```

Comment:
I believe the code in this function uses this so it likely should not start with an underscore as that is a convention to indicate it's not used.

Reply:
@Omega359 clippy was failed with 

`Parameter config is only used in recursion. Consider prefixing it with an underscore: _config.`

Reply:
Ahh. My apologies, I didn't see that